Leading indicators dap

kateharwood opened the pull request:
PR for the LeadingIndicatorsTools that will be used to create the notebook(s) for the DAP final product.

kateharwood committed:
adding leading indicators DAP

kateharwood committed:
adding leading indicators DAP

kateharwood committed:
adding a tools script for functions

kateharwood committed:
adding a tools script for functions

vshankar1 committed:
Add files via upload

Please find a revised version of LeadingIndicatorTools
(1) get_and_prepare_signals has been revised to include the 0/-ve cases. I have fixed a bug in the code (at-least I thought it was) regarding the indexing, and have commented out the lines to read in data from covidcast for now. 
(2) identify_beginning_increase_points is a new function, which takes in the raw covidcast signals and then returns a list of dataframes with a column that specifies points at the beginning of a rise in cases or indicator.
(3) plot_smoothed_signals_increase_point is a new function that plots the smoothed version of the signals and highlights the points identified in (2). Some further work needs to be done on re-scaling the axes.

vshankar1 committed:
Add files via upload

Please find a revised version of LeadingIndicatorTools
(1) get_and_prepare_signals has been revised to include the 0/-ve cases. I have fixed a bug in the code (at-least I thought it was) regarding the indexing, and have commented out the lines to read in data from covidcast for now. 
(2) identify_beginning_increase_points is a new function, which takes in the raw covidcast signals and then returns a list of dataframes with a column that specifies points at the beginning of a rise in cases or indicator.
(3) plot_smoothed_signals_increase_point is a new function that plots the smoothed version of the signals and highlights the points identified in (2). Some further work needs to be done on re-scaling the axes.

kateharwood committed:
adding V3 and test notebook

kateharwood committed:
adding V3 and test notebook

kateharwood committed:
cleaning up old files

kateharwood committed:
cleaning up old files

benjaminysmith commented on leading-indicators-dap/LeadingIndicatorTools.R:
```diff
@@ -0,0 +1,332 @@
+#
+# OUTPUT
+# @return A list including: cases = list of dataframes for cases by location, indicator = list of dataframes for indicator by location
+get_and_prepare_signals <- function(start_day, end_day, indicator_source, indicator_signal, 
```
"prepare" is vague -- what specifically is being done?

kateharwood replied:
Will revise this.

benjaminysmith commented on leading-indicators-dap/LeadingIndicatorTools.R:
```diff
@@ -0,0 +1,332 @@
+get_and_prepare_signals <- function(start_day, end_day, indicator_source, indicator_signal, 
+                                   case_threshold, indicator_threshold, geo_type = "county") {
+  # Get cases and indicator
+  cases = covidcast_signal(data_source = "usa-facts", signal = "confirmed_7dav_incidence_num",
```
Should we make the case source a parameter to the parent too? Right now this is locked to JHU.

kateharwood replied:
Sure I can do that.

benjaminysmith commented on leading-indicators-dap/LeadingIndicatorTools.R:
```diff
@@ -0,0 +1,332 @@
+  
+  # Split CASES into lists by geo
+  geos_case = cases %>%  group_keys(geo_value) %>% unlist()
+  case_list = cases %>% group_by(geo_value) %>% select(geo_value, time_value, value) %>% arrange(time_value) %>% group_split() %>% setNames(geos_case)
```
nit, but I think for readability it helps having each operation on a separate line. Did you autoformat in Rstudio?

kateharwood replied:
Will do.

benjaminysmith commented on leading-indicators-dap/LeadingIndicatorTools.R:
```diff
@@ -0,0 +1,332 @@
+
+
+
+# Get increase points
```
expand on "Increase points" (it's not clear what this means)

kateharwood replied:
I was just leading every function contract with the name of the function and letting the input and output describe what the function does, but I can add a more detailed description to this one.

benjaminysmith commented on leading-indicators-dap/LeadingIndicatorTools.R:
```diff
@@ -0,0 +1,332 @@
+#                 time_value, geo_value, case_value, ind_value, case_rise_point, indicator_rise_point
+get_increase_points <- function(case_list, indicator_list)
+{
+  final_list_with_leading_indicator=vector("list", length(case_list))
```
"final_list" is confusing -- why is it final?

kateharwood replied:
I think this was an artifact from a previous version of this function, will rename.

benjaminysmith commented on leading-indicators-dap/LeadingIndicatorTools.R:
```diff
@@ -0,0 +1,332 @@
+get_increase_points <- function(case_list, indicator_list)
+{
+  final_list_with_leading_indicator=vector("list", length(case_list))
+  for(i in 1:length(case_list))
```
I'm not good enough at R to suggest an alternative -- but is there a matrixy way to do this instead of using a for loop?

kateharwood replied:
I don't think I am the best person for that either. @vshankar1 @sangwon-hyun?

benjaminysmith commented on leading-indicators-dap/LeadingIndicatorTools.R:
```diff
@@ -0,0 +1,332 @@
+}
+
+
+# Get recall strict
```
Is there a more specific description for "strict" you can use?

kateharwood replied:
Yeah, I was figuring once we decided how we actually want to do recall and precision these functions will change. Named it strict for now just because I think this is stricter criteria than we probably want to go with in the end.

benjaminysmith replied:
Maybe add a TODO

benjaminysmith commented on leading-indicators-dap/LeadingIndicatorTools.R:
```diff
@@ -0,0 +1,332 @@
+}
+
+
+# TODO
```
update?

kateharwood replied:
Was leaving this for @vshankar1, I think he can provide a better exact description of the input and what this function returns than I can.

vshankar1 commented on leading-indicators-dap/LeadingIndicatorTools.R:
```diff
@@ -0,0 +1,332 @@
+      while(success && k <= length(case_dates)) {
+        success = FALSE
+        for (j in (1:length(indicator_dates))) {
+          if(case_dates[k] - indicator_dates[j] <= success_window && case_dates[k] - indicator_dates[j] > 3) {
```
Where does 3 come from in this line?

kateharwood replied:
Ah good call I didn't put that in the function contract, but was noticing there were a lot of examples when the leading indicator rise point was basically right on top of the case rise point, and a 1 or 2 day "leadingness" doesn't really seem like a good example of actual leadingness. But 3 is arbitrary - I will make that a param.

kateharwood committed:
addressing PR comments

kateharwood committed:
fixing merge conflicts

kateharwood committed:
start to final DAP notebook

kateharwood committed:
adding signals to section 3

kateharwood committed:
adding freq distributions

vshankar1 committed:
Add files via upload

vshankar1 committed:
Add files via upload

kateharwood committed:
updating LeadingIndicatorTools with new params for rise points and new recall and precision functions

vshankar1 committed:
Add files via upload

vshankar1 committed:
Update LeadingIndicatorTools.R

benjaminysmith commented:
Should I re-review this yet?

kateharwood commented:
Ben, the PR is here
https://github.com/cmu-delphi/covidcast-modeling/pull/30/files# if you want
to check it out.

On Thu, Mar 25, 2021 at 4:50 PM Ben Smith ***@***.***> wrote:

> Should I re-review this yet?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cmu-delphi/covidcast-modeling/pull/24#issuecomment-807462938>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADMIOEARGENJJGGT2C3SOELTFOO27ANCNFSM4WDDYRCA>
> .
>

kateharwood committed:
adding in Justin's edits to notebook and fixes to LeadingIndicatorTools

kateharwood committed:
Merge branch 'leading-indicators-dap' of https://github.com/cmu-delphi/covidcast-modeling into leading-indicators-dap

kateharwood committed:
updating tools file with comments addressed

kateharwood committed:
Merge branch 'leading-indicators-dap' into tools-file-update

kateharwood committed:
Merge pull request #31 from cmu-delphi/tools-file-update

Tools file update

kateharwood committed:
reformatting file and adding more helpful comments

kateharwood committed:
adding ideas for limitations

kateharwood committed:
renaming some params, variables, functions and updating comments

kateharwood committed:
rename function

kateharwood committed:
adding motivation and recall and precision methods to notebook

kateharwood committed:
updating case and indicator threshold functionality

kateharwood committed:
adding rest of notebook

kateharwood committed:
adding function sig

vshankar1 committed:
Update LeadingIndicatorsDAPNotebook.Rmd

Changes suggested/Things to Check
(1) Comparisons for numbers are on the same page
(2) Modify sentences in Motivation that this notebook follows 'directives'
(3) Occurrences is not spelled correctly for the leadingness histogram plots
(4) Edit the estimated derivative section to provide remove discussion of the various methods attempted (since we do not provide details on any)
(5) Provide motivation for county selection 
(6) Add F1 score to enable ranking of different signals. (2*precision*recall)/(precision+recall)
(7) Added brief conclusion

kateharwood committed:
Merge pull request #33 from cmu-delphi/leading-indicator-notebook-last-changes

Update LeadingIndicatorsDAPNotebook.Rmd

kateharwood committed:
changing wording and spelling

kateharwood committed:
merge conflicts

Commit:
last tweaks

kateharwood committed:
Merge pull request #32 from cmu-delphi/leading-indicators-dap-analysis

Adding rest of notebook
